lib/vehicle.rb

```diff
@@ -22,4 +27,18 @@ def antique?
   def electric_vehicle?
     @engine == :ev
   end
+
+  def registration_date
+    @registration_date
+  end
+
+  def plate_type
+    if @year <= Date.today.year.to_i - 25
+      @plate_type = :antique
+    elsif @engine == :ev
+      @plate_type = :ev
+    else 
+      @plate_type = :regular
+    end
+  end
```

Comment:
Is there value in having a `plate_type` method in addition to the `plate_type` attribute above, which has an attr_accessor?
Every time `vehicle_instance.plate_type` is called, it will run through this logic _and_ update the attribute. Maybe another method could be responsible for updating the `@plate_type` attribute, say, at the time of vehicle registration. That way when we call `vehicle_instance.plate_type`, we're actually calling the attribute directly.